docs/contributing.md

```diff
@@ -0,0 +1,327 @@
+  someone updates the version in the `pyproject.toml`).
+  - Known issues and fixes:
+    - If using `**kwargs`, you either have to specify all possible types of
+    `kwargs`, e.g. `**kwargs: Union[int, boolean]` or use `**kwargs: Any`
+- **`pytest`**: On GitHub Actions you can see which test failed. Reproduce it
+locally, e.g., using `pytest -n auto tests/linearGaussian_snpe_test.py`. Note
+that this will run for a few minutes and should result in passes and expected
+fails (xfailed).
+- Commit and push again until CI tests pass. Don't hesitate to ask for help by
+  commenting on the PR.
+
+#### mini-sbibm tests
+
+As SBI is a fundamentally data-driven approach, we are not only interested in whether
+the modifications to the codebase "pass the tests" but also in whether they improve or
+at least do not deteriorate the performance of the package for inference. To this end,
+we have a set of *mini-sbibm* tests (a minimal version of the sbi benchmarking package [`sbibm`](https://github.com/sbi-benchmark/sbibm)) that are intended for developers to run locally.
+
+These tests differ from the regular tests in that they always pass (provided there
+are no errors) but output performance metrics that can be compared, e.g., to the
+performance metrics of the main branch or relative to each other. The user-facing API
+is available via `pytest` through custom flags. To run the mini-sbibm tests, you can use
+the following command:
+
+```bash
+    pytest --bm
+```
+
+This will run all the mini-sbibm tests on all methods with default parameters and output
+the performance metrics nicely formatted to the console. If you have multiple CPU cores
+available, you can run the tests in parallel using the `-n auto` flag:
+
+```bash
+    pytest --bm -n auto
+```
+
+What if you are currently working on a specific method and you want to run the
+mini-sbibm tests only for this class of methods? You can use the `--bm-mode` flag:
+
+```bash
+    pytest --bm --bm-mode nspe
+```
+
+This will run the mini-sbibm tests only for methods of the `nspe` class, but with a
+few major hyperparameter choices, such as different base network architectures and
+different diffusion processes.
+
+The currently available modes are: `"npe"`, `"nle"`, `"nre"`, `"fmpe"`, `"npse"`,
+`"snpe"`, `"snle"`, and `"snre"`. If you require another mode, you can add it to the
+test suite in `tests/test_bm.py`.
+
+## Contributing to the documentation
+
+Most of the documentation for `sbi` is written in markdown and the website is generated
+using `mkdocs` with `mkdocstrings` and `mike`. The tutorials and examples are converted
+from jupyter notebooks into markdown files to be shown on the website. To work on
+improvements of the documentation, you should first  install the `doc` dependencies:
+
+```bash
+pip install -e ".[doc]"
+```
+
+Then, you can build the website locally by executing in the `docs` folder
+
+```bash
+mkdocs serve
+```
```

Comment:
outdated? change to sphinx / readthedocs?

Reply:
Since rtd is still hypernating (we do not yet advertise the website anywhere), I suggest to change this later.